### DISCUSSION TITLE
unist for CSS?

### DISCUSSION THREAD

**Discussion by justbyitself** [syntax-tree/unist#134] [Ideas] [answered]
It would be nice to create CSS ASTs in [unist](https://github.com/syntax-tree/unist). Is there any project like this? Something like [hast](https://github.com/syntax-tree/hast) but for CSS instead HTML. Or something like [css-tree](https://github.com/csstree/csstree) but for [unified](https://unifiedjs.com/) ecosystem. What do you think about it?

Thank you.

**Comment by wooorm** [comment]
Could be simple, like esast over estree: https://github.com/syntax-tree/esast. `css-tree` already uses types, so it would mainly be about `position` field (and maybe JSON compatibility?).

That being said, a “standard” without things using it isn’t very useful. Potentially even bad. So there’d need to be an ecosystem too.

**Reply by justbyitself** [reply]
I'm agree. 

The best of unified is its ecosystem. If CSSTree would use unist, I'm sure a nice ecosystem around would grow around soon. For example, It would be nice something like [hastscript](https://github.com/syntax-tree/hastscript) but for build CSS trees. And many little plugins could transform that trees in many ways.

Thank you for your answer.

**Reply by ChristianMurphy** [reply, marked as answer]
You can also see some past discussion and implementation here: https://github.com/unifiedjs/ideas/issues/3
